pyiron_base/project/generic.py

```diff
@@ -1609,10 +1610,11 @@ def register_tools(cls, name: str, tools):
             tools (Toolkit): The tools to register.
         """
         if hasattr(cls, name):
```

Comment:
Let's try rescue raising the error first.  Maybe something like this (untested though(

```suggestion
        if hasattr(cls, name) or str(type(getattr(cls, name))) == str(type(tools)):
```
and then raise the error inside the if again.

Reply:
Thinking again about it the condition would actually be something like
```python
if hasattr(cls, name) and str(type(getattr(cls, name))) != str(type(tools)):
```
Since in the case where the types match, we actually *want* to overwrite the tools set on the project (otherwise the reloaded version of the project will have the old version of the tools).